lib/btrfs_test.pm

```diff
@@ -114,6 +117,55 @@ sub cron_mock_lastrun {
     assert_script_run 'ls -al /var/spool/cron/lastrun/cron.{hourly,daily,weekly,monthly}';
 }
 
+# recent versions of snapper support json output
+# otherwise fallback to old implementation of parsing
+sub get_last_snap_number {
+    if (!defined($old_snapper)) {
+        $old_snapper = script_run('snapper --jsonout list --disable-used-space');
+    }
+
+    if (!!$old_snapper) {
+        return _get_last_snap_number_old();
+    }
+
+    my $snaps = decode_json(script_output('snapper --jsonout list --disable-used-space'));
+    my $last = (@{$snaps->{root}})[-1];
+    return $last->{number};
+}
+
+# In many cases script output returns not only script execution results
+# but other data which was written to serial device. We have to ensure
+# that we got what we expect. See poo#25716
+sub _get_last_snap_number_old {
+    # get snapshot id column, parse output in perl to avoid SIGPIPE
```

Comment:
This whole function can be done in a single awk call:

```
snapper list --disable-used-space | awk -F '|' 'NR == 1 { for(i=0; i < NF; ++i) if (match($i, /#/)) colno = i; } END { print strtonum($colno); }'
```

Reply:
I do not doubt that, but I am not sure whether we want to invest much time to code that is meant to be deleted. Maybe we can do that in another PR and merge this one for now

Reply:
That is the thing, the VRs are already done for now. If we need to add another set or another round then yes, I can replace that code

Reply:
> That is the thing, the VRs are already done for now.

The one for TW is red, at least `snapper_used_space` needs to be adjusted as well

Reply:
I think the used space column was missing in the basic list output. Lemme check it once more.